src/opm/parser/eclipse/EclipseState/Schedule/Well/WellMatcher.cpp

```diff
@@ -42,25 +57,25 @@ std::vector<std::string> WellMatcher::wells(const std::string& pattern) const {
 
     // WLIST
     if (pattern[0] == '*' && pattern.size() > 1)
-        return this->m_wlm.wells(pattern);
+       return this->sort_wells( this->m_wlm.wells(pattern) );
 
     // Normal pattern matching
     auto star_pos = pattern.find('*');
     if (star_pos != std::string::npos) {
         std::vector<std::string> names;
-        for (const auto& wname : this->m_wells) {
+        for (const auto& [wname, _] : this->m_wells) {
+            (void)_;
             int flags = 0;
             if (fnmatch(pattern.c_str(), wname.c_str(), flags) == 0)
                 names.push_back(wname);
         }
-        return names;
+        return this->sort_wells( std::move(names) );
     }
 
-    auto name_iter = std::find(this->m_wells.begin(), this->m_wells.end(), pattern);
+    auto name_iter = std::find_if(this->m_wells.begin(), this->m_wells.end(), [&pattern](const std::pair<std::string, std::size_t>& arg) { return arg.first == pattern; });
```

Comment:
I'm probably missing something.  `WellMatcher::m_wells` is an `unordered_map<string, size_t>` so why does this not use
```C++
auto name_iter = this->m_wells.find(pattern);
```
instead?

Reply:
Ehhh - you are right ;-)